### PR TITLE
chown support

### PR DESCRIPTION
**Pull request by sbollingerAtRocket**
Signed-off-by: Steven Bollinger <sbollinger@rocketsoftware.com>

### CODE REVIEW

File: c/unixFileService.c [hunk 1]
```diff
@@ -920,6 +920,38 @@ static int serveUnixFileMetadata(HttpService *service, HttpResponse *response) {
...
+  }
+  else {
+    jsonPrinter *out = respondWithJsonPrinter(response);
+
+    setResponseStatus(response, 405, "Method Not Allowed");
+    setDefaultJSONRESTHeaders(response);
+    addStringHeader(response, "Allow", "POST");
+    writeHeader(response);
+
+    jsonStart(out);
```
Comment (ifakhrutdinov) [inline c/unixFileService.c]: Do we really need this empty JSON? If not, we can get rid of the `respondWithJsonPrinter` call.

File: c/unixFileService.c [hunk 2]
```diff
@@ -920,6 +920,38 @@ static int serveUnixFileMetadata(HttpService *service, HttpResponse *response) {
   return 0;
 }
 
+static int serveUnixFileChangeOwner (HttpService *service, HttpResponse *response) {
+  HttpRequest *request = response->request;
+  char *routeFileFrag = stringListPrint(request->parsedFile, 2, 1000, "/", 0);
```
Comment (ifakhrutdinov) [inline c/unixFileService.c]: Can we check the input parameters? For example, `directoryChangeOwnerAndRespond` function uses `recursive` as is, without checking for NULL. Currently it's not clear which parameters are optional.

**Comment by ifakhrutdinov** [comment]
Can you pleas also link the pull-requests this pull-request depends on? Otherwise it may get merged without the corresponding changes in zowe-common-c.

File: c/unixFileService.c (hunk 1 above)
Comment (sbollingerAtRocket) [reply c/unixFileService.c]: The jira ticket specifies that we support the POST for this feature.  If you want to remove the message indicating what the is error in the request, then we can remove it.  Otherwise, I would keep it in.

File: c/unixFileService.c (hunk 2 above)
Comment (sbollingerAtRocket) [reply c/unixFileService.c]: The code uses strcmp to set the bool recursive.  strcmp will return false if the string is null.  We can add an additional test for null to make it more clear.

File: c/unixFileService.c (hunk 1 above)
Comment (ifakhrutdinov) [reply c/unixFileService.c]: I was talking about the lines:
```c
jsonStart(out);
jsonEnd(out);
```

File: c/unixFileService.c (hunk 2 above)
Comment (ifakhrutdinov) [reply c/unixFileService.c]: > strcmp will return false if the string is null. 

It may work or it may segfault/S0C4, I think it's implementation defined.

Comment (daveyc) [reply c/unixFileService.c]: On z/OS strcmp() will just invoke a builtin which boils down to a CLST instruction with no check for NULL. I suspect it will result in a 0C4.

Comment (daveyc) [reply c/unixFileService.c]: FWIW, I don't use strcmp! I prefer to use macros or inline functions that check for null. 

```
inline bool streq(const char * lhs, const char * rhs) {            
  return lhs != NULL && rhs != NULL && strcmp(lhs, rhs) == 0;      
}                                                                  
```

Comment (sbollingerAtRocket) [reply c/unixFileService.c]: I have put in the additional test for NULL before the strcmp so there shouldn't be a problem.

File: c/unixFileService.c (hunk 1 above)
Comment (sbollingerAtRocket) [reply c/unixFileService.c]: I can remove this is un-necessary.  Make sure it is;  there are 7 other places in this file with the exact code sequence.

Comment (ifakhrutdinov) [reply c/unixFileService.c]: You don't need to fix everything. If this is not necessary, why add more of it.

File: c/unixFileService.c (hunk 2 above)
Comment (1000TurquoisePogs) [reply c/unixFileService.c]: I see this here now: https://github.com/zowe/zowe-common-c/pull/104/files#diff-5a6b6901a39402924b29b573b54bbc0eR571
